mdata/cache/ccache_metric.go

```diff
@@ -345,6 +345,12 @@ func (mc *CCacheMetric) searchForward(ctx context.Context, metric schema.AMKey,
 	// add all consecutive chunks to search results, starting at the one containing "from"
 	for ; ts != 0; ts = mc.chunks[ts].Next {
 		log.Debugf("CCacheMetric searchForward: forward search adds chunk ts %d to start", ts)
+		_, ok := mc.chunks[ts]
```

Comment:
Afaict we currently don't fully understand the issue which has lead to this panic you're trying to prevent, so it's possible that there is some race condition which deleted an element from `mc.chunks` after `ts` was set. In the case of a race condition you couldn't be sure that this item with the key `ts` doesn't get deleted from `mc.chunks` between `:348` and `:354`, so i think it would be safer to take the reference to that chunk here and then use it everywhere further down instead of accessing `mc.chunks[ts]` again

Reply:
My original plan did just that, I changed it because I thought this might be more efficient. I will go take closer look at locking / unlocking and see if I can track anything down, if not I am fine changing this to retrieve the item at this point.

Reply:
i don't know why this would be more efficient. by taking the reference to the map entry and using it you can also avoid additional map key lookups

Reply:
if !ok it would be one less allocation, it doesn't much matter I guess since it is a pointer anyway.

I looked over the locking / unlocking and don't see how anything could modify mc.chunks yet while inside this function. I will update it to get the cached chunk for now. Obviously something IS happening, but I haven't tracked it down yet.